org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.java

```diff
@@ -0,0 +1,489 @@
+		button.setFont(button.getParent().getFont());
+		button.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		config.apply(button);
+		return button;
+	}
+
+	protected Button Radio(Composite composite, Procedure1<? super Button> config) {
+		Button button = new Button(composite, SWT.RADIO);
+		button.setFont(button.getParent().getFont());
+		button.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		config.apply(button);
+		return button;
+	}
+
+	protected Combo DropDown(Composite parent, Procedure1<? super Combo> config) {
+		Combo combo = new Combo(parent, SWT.READ_ONLY);
+		combo.setFont(parent.getFont());
+		combo.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		config.apply(combo);
+		return combo;
+	}
+
+	protected ControlDecoration InfoDecoration(Control control, String text) {
+		FieldDecoration infoField = FieldDecorationRegistry.getDefault().getFieldDecoration(FieldDecorationRegistry.DEC_INFORMATION);
+		ControlDecoration controlDecoration = new ControlDecoration(control, (SWT.TOP + SWT.RIGHT));
+		controlDecoration.setImage(infoField.getImage());
+		controlDecoration.setDescriptionText(text);
+		controlDecoration.setShowHover(true);
+		return controlDecoration;
+	}
+
+	protected void setDefaults() {
+		createUiProject.setSelection(true);
+		createIdeProject.setSelection(true);
+		createTestProject.setSelection(true);
+		createWebProject.setSelection(false);
+		createSDKProject.setSelection(false);
+		createP2Project.setSelection(false);
+
+		select(preferredBuildSystem, BuildSystem.NONE);
+		select(createLanguageServer, LanguageServer.NONE);
+		select(sourceLayout, SourceLayout.PLAIN);
+	}
+
+	public boolean isCreateUiProject() {
+		return createUiProject.getSelection();
+	}
+
+	public boolean isCreateTestProject() {
+		return createTestProject.getSelection();
+	}
+
+	public boolean isCreateIdeProject() {
+		return createIdeProject.getSelection();
+	}
+
+	public boolean isCreateWebProject() {
+		return createWebProject.getSelection();
+	}
+
+	public boolean isCreateSdkProject() {
+		return createUiProject.getSelection() && createSDKProject.getSelection();
+	}
+
+	public boolean isCreateP2Project() {
+		return createUiProject.getSelection() && createP2Project.getSelection();
+	}
+
+	public BuildSystem getPreferredBuildSystem() {
+		return BuildSystem.values()[preferredBuildSystem.getSelectionIndex()];
+	}
+
+	public SourceLayout getSourceLayout() {
+		return SourceLayout.values()[sourceLayout.getSelectionIndex()];
+	}
+
+	public LanguageServer getLanguageServer() {
+		if (getPreferredBuildSystem() == BuildSystem.NONE) {
+			return LanguageServer.NONE;
+		}
+		return LanguageServer.values()[createLanguageServer.getSelectionIndex()];
+	}
+
+	public JUnitVersion getSelectedJUnitVersion() {
+		if (junitVersion4.getSelection()) {
+			return JUnitVersion.JUNIT_4;
+		} else if (junitVersion5.getSelection()) {
+			return JUnitVersion.JUNIT_5;
+		} else {
+			throw new IllegalStateException();
```

Comment:
Might be nice to add a message to this exception e.g. "JUnit version could not be determined"